.github/workflows/node.js.yml

```diff
@@ -0,0 +1,26 @@
+name: Node.js CI
+
+on:
+  pull_request:
+    branches: 
+      - "development"
+
+jobs:
+  type-check-lint-test-build:
+    name: type-check-lint-test-build
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Use Node.js 18.x
+      uses: actions/setup-node@v3
+      with:
+        node-version: "18.x"
+        cache: 'npm'
+    - uses: nrwl/nx-set-shas@v3
+    - run: npm install nx
```

Comment:
You can give a name to each step to debug more easily if sth is broken in the ci.
`
    - name: install Dependencies
        uses: npm i
        
      - name: Run Linting Rules
        run: npm run lint
`
and like so.